src/api/block_builder/input_selection/sender_issuer.rs

```diff
@@ -0,0 +1,260 @@
+                        let output_response = self.client.get_output(&output_id).await?;
+                        if let OutputDto::Nft(nft_output) = &output_response.output {
+                            let nft_output = NftOutput::try_from(nft_output)?;
+
+                            let unlock_address = nft_output
+                                .unlock_conditions()
+                                .locked_address(nft_output.address(), current_time);
+
+                            let (address_index, internal) = match self.secret_manager {
+                                Some(secret_manager) => {
+                                    match unlock_address {
+                                        Address::Ed25519(_) => {
+                                            search_address(
+                                                secret_manager,
+                                                &bech32_hrp,
+                                                self.coin_type,
+                                                self.account_index,
+                                                self.input_range.clone(),
+                                                unlock_address,
+                                            )
+                                            .await?
+                                        }
+                                        // Alias and NFT addresses can't be generated from a private key
+                                        _ => (0, false),
+                                    }
+                                }
+                                None => (0, false),
+                            };
+
+                            required_inputs.push(InputSigningData {
+                                output: Output::try_from(&output_response.output)?,
+                                output_metadata: OutputMetadata::try_from(&output_response.metadata)?,
+                                chain: Some(Chain::from_u32_hardened(vec![
+                                    HD_WALLET_TYPE,
+                                    self.coin_type,
+                                    self.account_index,
+                                    internal as u32,
+                                    address_index,
+                                ])),
+                                bech32_address: unlock_address.to_bech32(&bech32_hrp),
+                            });
+                        }
+                    }
+                }
+            }
+        }
+
+        // Check required alias and nft outputs with new added outputs
+        // no need to check for sender and issuer again, since these outputs already exist and we don't set new features
+        // for them
+        let utxo_chain_inputs = self
+            .get_utxo_chains_inputs(
+                &required_inputs
+                    .iter()
+                    .map(|i| i.output.clone())
+                    .collect::<Vec<Output>>(),
```

Comment:
`get_utxo_chains_inputs` can probably accept a `impl Iterator<Item = &'a Output> + Clone`, removing the clones and collect